packages/contracts-bedrock/src/L2/L1Block.sol

```diff
@@ -212,4 +215,67 @@ contract L1Block is ISemver {
             sstore(operatorFeeConstant.slot, shr(160, calldataload(164)))
         }
     }
+
+    /// @notice Updates the L1 block values for a Jovian upgraded chain.
+    /// Params are packed and passed in as raw msg.data instead of ABI to reduce calldata size.
+    /// Params are expected to be in the following order:
+    ///   1. _baseFeeScalar         L1 base fee scalar
+    ///   2. _blobBaseFeeScalar     L1 blob base fee scalar
+    ///   3. _sequenceNumber        Number of L2 blocks since epoch start.
+    ///   4. _timestamp             L1 timestamp.
+    ///   5. _number                L1 blocknumber.
+    ///   6. _basefee               L1 base fee.
+    ///   7. _blobBaseFee           L1 blob base fee.
+    ///   8. _hash                  L1 blockhash.
+    ///   9. _batcherHash           Versioned hash to authenticate batcher by.
+    ///   10. _operatorFeeScalar    Operator fee scalar.
+    ///   11. _operatorFeeConstant  Operator fee constant.
+    ///   12. _daFootprintGasScalar DA footprint gas scalar.
+    function setL1BlockValuesJovian() public {
+        _setL1BlockValuesJovian();
+    }
+
+    /// @notice Updates the L1 block values for a Jovian upgraded chain.
+    /// Params are packed and passed in as raw msg.data instead of ABI to reduce calldata size.
+    /// Params are expected to be in the following order:
+    ///   1. _baseFeeScalar         L1 base fee scalar
+    ///   2. _blobBaseFeeScalar     L1 blob base fee scalar
+    ///   3. _sequenceNumber        Number of L2 blocks since epoch start.
+    ///   4. _timestamp             L1 timestamp.
+    ///   5. _number                L1 blocknumber.
+    ///   6. _basefee               L1 base fee.
+    ///   7. _blobBaseFee           L1 blob base fee.
+    ///   8. _hash                  L1 blockhash.
+    ///   9. _batcherHash           Versioned hash to authenticate batcher by.
+    ///   10. _operatorFeeScalar    Operator fee scalar.
+    ///   11. _operatorFeeConstant  Operator fee constant.
+    ///   12. _daFootprintGasScalar DA footprint gas scalar.
+    function _setL1BlockValuesJovian() internal {
```

Comment:
The implementation in this function may be overkill. Worth taking a closer look here.